### PR TITLE
feat: allow custom login and logout paths

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Description

Allows custom login and logout paths in the same way as custom callback ones. See API.md for more information. Custom routes can also be done [this way](https://github.com/auth0/express-openid-connect/blob/master/EXAMPLES.md#example-2).

### Testing

- [x] This change adds test coverage for new/changed/fixed functionality

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -27,12 +27,15 @@ In general, you won't need to configure this middleware besides the required par
...
+| required            | `true`                          | If true requires authentication for all the routes in the stack. You can also provide a function to determine if is required based on the request.               |
 | handleUnauthorizedErrors | `true`                     | Install a middleware that handles Unauthorized/401 errors by triggering the login process. |
 | routes              | `true`                          | Installs the `GET /login` and `GET /logout` route.                              |
 | idpLogout           | `false`                         | Logout the user from the identity provider on logout                            |
 | auth0Logout         | `false`                         | Enable Auth0's non-compliant logout feature, only if Auth0 can be detected and the Auth0 instance does not support OpenID Connect session management. |
-| authorizationParams | See bellow                      | The parameters for the authorization call.                                      |
+| redirectUriPath     | `/callback`                     | Relative path for the callback URL; sent to the authroize endpoint in the `redirectUri` parameter. |
```
Comment (lbalmaceda) [inline API.md]:
```suggestion
| redirectUriPath     | `/callback`                     | Relative path for the callback URL; sent to the authorize endpoint in the `redirectUri` parameter. |
```

File: API.md [hunk 2]
```diff
@@ -27,12 +27,15 @@ In general, you won't need to configure this middleware besides the required par
 | clientSecret        | `env.CLIENT_SECRET`             | The client secret, only required for some grants.                              |
 | clockTolerance      | `5`                             | The clock's tolerance in seconds for token verification.                       |
 | getUser             | `tokenSet => tokenSet.claims()` | An async function receiving a tokenset and returning the profile for `req.openid.user`. |
-| required            | `true`                            | If true requires authentication for all the routes in the stack. You can also provide a function to determine if is required based on the request.               |
+| required            | `true`                          | If true requires authentication for all the routes in the stack. You can also provide a function to determine if is required based on the request.               |
 | handleUnauthorizedErrors | `true`                     | Install a middleware that handles Unauthorized/401 errors by triggering the login process. |
 | routes              | `true`                          | Installs the `GET /login` and `GET /logout` route.                              |
```
Comment (lbalmaceda) [inline API.md]: wouldn't this change with this PR as well?

File: lib/config.js [hunk 3]
```diff
@@ -29,6 +29,8 @@ const paramsSchema = Joi.object().keys({
   errorOnRequiredAuth: Joi.boolean().optional().default(false),
   auth0Logout: Joi.boolean().optional().default(false),
   redirectUriPath: Joi.string().optional().default('/callback'),
+  loginPath: Joi.string().optional().default('/login'),
```
Comment (lbalmaceda) [inline lib/config.js]: I think you're assuming the `/` is always included in the value, shouldn't you validate that?

Comment (joshcanhelp) [reply lib/config.js]: Probably not a bad idea. I'm using the same convention as the callback but it's probably good to do for all 3. Good catch.

#21

File: API.md (hunk 2 above)
Comment (joshcanhelp) [reply API.md]: No. This switch makes authentication required on all routes when it's `true` (default but considering whether that's best) or not (using `requireAuth()`) when it's `false`.

Comment (lbalmaceda) [reply API.md]: I mean the description of the property. If the /login and /logout paths can now be customized, maybe it's worth pointing out those new properties refer to that value?

File: middleware/auth.js [hunk 4]
```diff
@@ -68,10 +68,10 @@ module.exports = function (params) {
   });
 
   if (config.routes) {
-    router.get('/login', (req, res) => {
+    router.get(config.loginPath, (req, res) => {
```
Comment (lbalmaceda) [inline middleware/auth.js]: I think it's missing the tests for this file changes. You are now testing the config options are being set, but you're not testing that they are being used in the router.

**Comment by joshcanhelp** [comment]
@lbalmaceda - Ready to re-review when you're able. Thank you!

**Pull request merged**